choco-solver/src/main/java/org/chocosolver/solver/Solver.java

```diff
@@ -873,8 +889,15 @@ public void findOptimalSolution(ResolutionPolicy policy, RealVar objective, doub
         if (!getObjectiveManager().isOptimization()) {
             set(new ObjectiveManager<RealVar, Double>(objective, policy, precision, true));
         }
-        set(new LastSolutionRecorder(new Solution(), true, this));
-        solve(false);
+        ISolutionRecorder oldRecorder = getSolutionRecorder();
```

Comment:
I didn't want to leave a stopped LastSolutionRecorder attached to the solver so I gave it back it's old one. Perhaps I should remove lines 894 and 898 and not touch the solvers solutionRecorder field.